### PR TITLE
Update idealstate when replication is changed for a table.

### PR DESCRIPTION
**Pull request by mcvsubbu**
When we change the replication for a table, we need to update the idealstate as well,
so that helix sends the right number of messages to the serving instances. Also,
a subsequent rebalance will do the right thing with the number of replicas per partition

### CODE REVIEW

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotTableIdealStateBuilder.java [hunk 1]
```diff
@@ -206,6 +206,8 @@ public static void buildLowLevelRealtimeIdealStateFor(String realtimeTableName,
     if (idealState == null) {
       idealState = buildEmptyKafkaConsumerRealtimeIdealStateFor(realtimeTableName, nReplicas);
       create = true;
+    } else {
+      idealState.setReplicas(Integer.toString(nReplicas));
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotTableIdealStateBuilder.java]: I think the nReplicas will be set in PinotLLCRealtimeSegmentManager.java updateIdealState() line 371, can you please check that? If yes, revert the change for this file.

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotHelixResourceManager.java [hunk 2]
```diff
@@ -1139,6 +1140,18 @@ public void setExistingTableConfig(AbstractTableConfig config, String tableNameW
     } else if (type == TableType.OFFLINE) {
```
Comment (jfim) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotHelixResourceManager.java]: Should this also apply to realtime tables?

Comment (mcvsubbu) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotHelixResourceManager.java]: For HLC, no because we have only one replica per segment.

For LLC, I tried to make a change but then backed it out as per comments, because a much bigger change is needed to increase number of replicas.
